server/src/main/java/io/unitycatalog/server/security/SecurityConfiguration.java

```diff
@@ -0,0 +1,119 @@
+package io.unitycatalog.server.security;
+
+import com.auth0.jwt.algorithms.Algorithm;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.security.KeyFactory;
+import java.security.KeyPair;
+import java.security.KeyPairGenerator;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.security.interfaces.RSAPrivateKey;
+import java.security.interfaces.RSAPublicKey;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.security.spec.X509EncodedKeySpec;
+import java.util.Base64;
+import lombok.SneakyThrows;
+import org.apache.commons.codec.binary.Hex;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Security settings for the Authnz framework.
+ *
+ * <p>The settings are loaded from files persisted in the UC configuration folder. If no settings
+ * files exist, they will be generated for reuse across restarts. Equivalent openssl commands are -
+ * openssl genrsa -out private_key.pem 2048 - openssl rsa -in private_key.pem -pubout -outform DER
+ * -out public_key.der - openssl pkcs8 -topk8 -inform PEM -outform DER -in private_key.pem -out
+ * private_key.der -nocrypt - openssl rand -hex -out key_id.txt 32
+ */
+public class SecurityConfiguration {
+
+  private static final Logger log = LoggerFactory.getLogger(SecurityConfiguration.class);
+
+  private Path rsa512PublicKey;
+
+  private Path rsa512PrivateKey;
+
+  private Path keyId;
+
+  private String rsa512PublicKeyEncoded = null;
+
+  private String rsa512PrivateKeyEncoded = null;
+
+  @SneakyThrows
+  public SecurityConfiguration(Path configurationFolder) {
+    rsa512PublicKey = configurationFolder.resolve("public_key.der");
+    rsa512PrivateKey = configurationFolder.resolve("private_key.der");
+    keyId = configurationFolder.resolve("key_id.txt");
+
+    initializeIfMissing();
+  }
+
+  @SneakyThrows
+  public void initializeIfMissing() {
+    if (Files.notExists(rsa512PublicKey)
+        || Files.notExists(rsa512PrivateKey)
+        || Files.notExists(keyId)) {
+      log.info("Initializing security configuration.");
+      KeyPairGenerator keyPairGenerator = KeyPairGenerator.getInstance("RSA");
+      keyPairGenerator.initialize(2048);
+      KeyPair keyPair = keyPairGenerator.generateKeyPair();
+
+      Files.write(rsa512PublicKey, keyPair.getPublic().getEncoded(), StandardOpenOption.CREATE);
+      Files.write(rsa512PrivateKey, keyPair.getPrivate().getEncoded(), StandardOpenOption.CREATE);
+
+      byte[] keyIdBytes = new byte[32];
+      new SecureRandom().nextBytes(keyIdBytes);
+      Files.writeString(keyId, Hex.encodeHexString(keyIdBytes), StandardOpenOption.CREATE);
+    }
+  }
+
+  public Algorithm algorithmRSA()
+      throws NoSuchAlgorithmException, InvalidKeySpecException, IOException {
+    return Algorithm.RSA512(rsaPublicKey(), rsaPrivateKey());
```

Comment:
Nit: What happens if the public key and private key don't match? This can happen if there's a race in reading the key files in `rsaPublicKey` and `rsaPrivateKey` while they are being updated (see my comment below)? Arguably this will be very rare so this PR can go in without fixing it, but would like to understand the impact (and maybe create a ticket to fix later)? 

The same kind of inconsistency can also happen in lines 57-60 of `SecurityContext` when the private key, public key, and key id are being read separately from the filesystem.

Reply:
Although there is technically a race condition, currently the keys are only created on startup, so in practicality that should never happen. If we built in a method to do manual or automatic rotation that could become a problem. I'll create an issue to keep note of it.

Reply:
If keys are only created on startup, the `rsaPublicKey`, `rsaPrivateKey`, and `getKeyId` can be simplified by reading them on startup, correct?